### PR TITLE
Fixing bug with sku index attribute

### PR DESCRIPTION
**Pull request by wiewiurdp**
# Description

Fixing bug with sku index attribute

Issue #1567  
## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Unit Tests
- [x] Behat Test

# Checklist:
- [x] I have read the contribution requirements and fulfill them.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: module/attribute/src/Application/Validator/AttributeCodeValidator.php [hunk 1]
```diff
@@ -50,8 +51,12 @@ public function validate($value, Constraint $constraint): void
             return;
         }
 
-        if (!preg_match($constraint->pattern, $value)) {
+        if (!preg_match($constraint->pattern, $value) || 0 === strpos(
+            $value,
+            SystemAttributeCode::SYSTEM_ATTRIBUTE_PREFIX
```
Comment (piotrkreft) [inline module/attribute/src/Application/Validator/AttributeCodeValidator.php:56]: Can't we extend AttributeCode value object to deny possibility of creating with `esa_` prefix?

**Review by piotrkreft — CHANGES_REQUESTED** [review]

File: module/attribute/src/Domain/ValueObject/AttributeCode.php [hunk 2]
```diff
@@ -13,7 +13,11 @@
 
 class AttributeCode extends AbstractCode
 {
-    public const PATTERN = '/^([a-zA-Z0-9_]+)$/';
+    public const NOT_SYSTEM_ATTRIBUTE_PATTERN = '/^(?!'.self::SYSTEM_ATTRIBUTE_PREFIX.')[a-zA-Z0-9_]+$/';
```
Comment (piotrkreft) [inline module/attribute/src/Domain/ValueObject/AttributeCode.php:16]: it's just a pattern as it was. SystemAttributeCode should have it's own definition and validate by it

File: module/attribute/tests/Application/Serializer/Normalizer/AttributeNormalizerTest.php [hunk 3]
```diff
@@ -21,23 +22,37 @@ class AttributeNormalizerTest extends TestCase
...
+
     protected function setUp(): void
     {
         $this->mockNormalizer = $this->createMock(NormalizerInterface::class);
+        $this->mockDenormalizer = $this->createMock(DenormalizerInterface::class);
 
         $this->normalizer = new AttributeNormalizer();
         $this->normalizer->setNormalizer($this->mockNormalizer);
+        $this->normalizer->setDenormalizer($this->mockDenormalizer);
+        $this->data['code'] = 'esa_system_attribute';
```
Comment (piotrkreft) [inline module/attribute/tests/Application/Serializer/Normalizer/AttributeNormalizerTest.php:46]: Runtime test specific data should not be set as properties

File: module/attribute/tests/Application/Serializer/Normalizer/AttributeNormalizerTest.php [hunk 4]
```diff
@@ -71,6 +85,37 @@ public function testShouldNotSupportNormalization($data): void
         $this->assertFalse($this->normalizer->supportsNormalization($data));
     }
 
+    public function testShouldDenormalizeAttribute(): void
```
Comment (piotrkreft) [inline module/attribute/tests/Application/Serializer/Normalizer/AttributeNormalizerTest.php:88]: Missing check of appropriate code setting for system and not system attribute

**Review by piotrkreft — CHANGES_REQUESTED** [review]

**Review by piotrkreft — APPROVED** [review]

**Review by wfajczyk — APPROVED** [review]

**Pull request merged**